packages/components/src/internal/components/chart/utils.ts

```diff
@@ -49,6 +50,101 @@ export function processChartData(response: ISelectRowsResult, options?: ProcessC
     return { barFillColors, data };
 }
 
+interface PercentageBarProps {
+    queryKey: string,
+    name: string;
+    label: string,
+    useForSubtitle?: boolean,
+    className?: string,
+    appURL?: AppURL,
+    filled?: boolean,
+}
+
+export function createPercentageBarData(
+    row: Record<string, any>,
+    itemNounPlural: string,
+    unusedLabel: string,
+    totalCountKey: string,
+    percentageBars: PercentageBarProps[],
+    baseAppURL?: AppURL,
+    urlFilterKey?: string
+) : { data: HorizontalBarData[], subtitle: string } {
+    const totalCount = caseInsensitive(row, totalCountKey)?.value ?? 0;
+    let unusedCount = totalCount;
+    const data = [];
+    let subtitle;
+    if (totalCount > 0) {
+        percentageBars.forEach(percentageBar => {
+            const itemCount = caseInsensitive(row, percentageBar.queryKey)?.value ?? 0;
+            unusedCount = unusedCount - itemCount;
+            const itemPct = (itemCount/totalCount) * 100;
+            const title = `${itemCount.toLocaleString()} of ${totalCount.toLocaleString()} ${itemNounPlural.toLowerCase()} are ${percentageBar.label.toLowerCase()}`;
+
+            data.push({
+                title,
+                name: percentageBar.name,
+                className: percentageBar.className,
+                count: itemCount,
+                totalCount: totalCount,
+                percent: itemPct,
+                href: percentageBar.appURL?.toHref() ?? baseAppURL?.addFilters(Filter.create(urlFilterKey, percentageBar.label)).toHref(),
+                filled: percentageBar.filled ?? true,
+            });
+
+            if (percentageBar.useForSubtitle) {
+                subtitle = `${title} (${(itemPct > 0 && itemPct < 1 ? '< 1' :  Math.trunc(itemPct))}%)`;
+            }
+        });
+
+        if (unusedLabel) {
+            const unusedPct = (unusedCount/totalCount) * 100;
+            const unusedTitle = `${unusedCount.toLocaleString()} of ${totalCount.toLocaleString()} ${itemNounPlural.toLowerCase()} are ${unusedLabel.toLowerCase()}`;
+            data.push({
+                title: unusedTitle,
+                name: 'Available',
+                count: unusedCount,
+                totalCount: totalCount,
+                percent: unusedPct,
+                filled: false,
+            });
+
+            if (!subtitle) {
+                subtitle = `${unusedTitle} (${(unusedPct > 0 && unusedPct < 1 ? '< 1' :  Math.trunc(unusedPct))}%)`;
+            }
+        }
+    }
+
+    return { data, subtitle };
+}
+
+export interface HorizontalBarLegendData {
+    circleColor: string;
+    backgroundColor: string;
+    legendLabel: string;
+}
+
+export function createHorizontalBarLegendData(data: HorizontalBarData[]) : HorizontalBarLegendData[] {
+    let legendMap = {};
+    data.forEach(sampleType => {
```

Comment:
Could be renamed to something not sample-specific.

Reply:
Thank you. I missed that. Renamed.